### PR TITLE
Enforce strict-raw-types

### PR DESCRIPTION
**Pull request by srawlins**
Enabling [strict-raw-types](https://github.com/dart-lang/language/blob/master/resources/type-system/strict-raw-types.md) leads to changes in a few categories:

* use `var` for inference
* put a type on a raw `Future`
* put a bound on a type parameter like `Object`
* put types on a raw `Map`. Cringe. This one doesn't look pretty, but can be helped once we migrate to null safety with the use of non-function type aliases.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ analyzer:
     todo: ignore
     unused_import: warning
     unused_shown_name: warning
+  language:
```
Comment (jcollins-g) [inline analysis_options.yaml:10]: This probably should be added to analysis_options_presubmit.yaml?

**Review by jcollins-g — APPROVED** [review]
Looks good, yeah, I'm not a fan of the explicit Object everywhere but I like how `strict-raw-types` seems to eliminate the ambiguity about what is derived vs implicit dynamic.

File: analysis_options.yaml (hunk 1 above)
Comment (srawlins) [reply analysis_options.yaml:10]: Done.

**Comment by srawlins** [comment]
Yeah `T extends Object` definitely makes my head scratch, for now. It looks like it is indicating some important grand distinction, when really it's just to avoid dynamic. I hope it just becomes more familiar.

**Pull request merged**